page.codeberg.gwync.glerm.yml

```diff
@@ -0,0 +1,30 @@
+id: page.codeberg.gwync.glerm
+runtime: org.freedesktop.Platform
+runtime-version: '24.08'
+sdk: org.freedesktop.Sdk
+command: glerm
+finish-args:
+  - --filesystem=home
```

Comment:
why does it need home?

Reply:
High score file. Is there a way to restrict to one filename?

Reply:
What is it for? It should probably be in an XDG location.

Reply:
Reading and writing ~/.glerm_score.

Reply:
No I mean what's the directory for? Do you store config files, cache or what?

Reply:
It's a single text file. If not extant, created, needs to be rw.

Reply:
So maybe --filesystem=~/.glerm_score:rw ?

Reply:
No instead modify your code to create it relative to $XDG_CACHE_HOME, $XDG_CONFIG_HOME (whichever is appropriate) and fallback to $HOME if those environment variables exist.

This will make it work in Flatpak without any extra permissions and also outside Flatpak.

Reply:
also saving dotfiles in the HOME directory like that is considered harmful and has been deprecate for a very long time. It should be consider to be a bug.

Reply:
by "a very long time" it's at least 20 years.

Reply:
> So maybe --filesystem=~/.glerm_score:rw ?

The documentation tells pretty clearly that it expects a directory here.

Reply:
Not actually clear to me, but I'll make that change. I hadn't encountered that dotfile practice but it makes sense.

I'll update my code.

Reply:
> No instead modify your code to create it relative to $XDG_CACHE_HOME or $XDG_CONFIG_HOME (whichever is appropriate) and fallback to $HOME/.config if those environment variables does not exist.
> 
> This will make it work in Flatpak without any extra permissions and also outside Flatpak.

Will this allow use of the same file by the native and flatpak versions of the app?

Reply:
No, why should it? People will install any one of the packages and stick with that. 

That'd defeat any sandboxing.

Reply:
It's not unheard of.  Would something like --filesystem=$HOME/.config/glerm/ work for that purpose?

Reply:
No, people who will migrate is an edge case and if they are technical enough to want to move their config files around, they can already copy it from home to `~/.var/app/$app-id/config`.

Reply:
Fixed.